Maple2.File.IO/Nif/Endian.cs

```diff
@@ -0,0 +1,146 @@
+﻿using System.Buffers.Binary;
+using System.Numerics;
+using System.Text;
+
+namespace Maple2.File.IO.Nif;
+
+public class EndianReader {
+    public bool Swap { get; init; }
+    public byte[] Data { get; init; }
+    public int Index { get; private set; }
+
+    public EndianReader(byte[] data, bool swap, int index = 0) {
+        Swap = swap;
+        Data = data;
+        Index = index;
+    }
+
+    public bool ReadBool() {
+        bool value = BitConverter.ToBoolean(Data, Index);
+
+        Index += 1;
+
+        return value;
+    }
+
+    public byte ReadByte() {
+        byte value = Data[Index];
+
+        Index += 1;
+
+        return value;
+    }
+
+    public ushort ReadUInt16() {
+        ushort value = BitConverter.ToUInt16(Data, Index);
+
+        Index += 2;
+
+        if (Swap) {
+            return BinaryPrimitives.ReverseEndianness(value);
+        }
+
+        return value;
+    }
+
+    public uint ReadUInt32() {
+        uint value = BitConverter.ToUInt32(Data, Index);
+
+        Index += 4;
+
+        if (Swap) {
+            return BinaryPrimitives.ReverseEndianness(value);
+        }
+
+        return value;
+    }
+
+    public int ReadInt32() {
+        int value = BitConverter.ToInt32(Data, Index);
+
+        Index += 4;
+
+        if (Swap) {
+            return BinaryPrimitives.ReverseEndianness(value);
+        }
+
+        return value;
+    }
+
+    public ulong ReadUInt64() {
+        ulong value = BitConverter.ToUInt64(Data, Index);
+
+        Index += 8;
+
+        if (Swap) {
+            return BinaryPrimitives.ReverseEndianness(value);
+        }
+
+        return value;
+    }
+
+    public float ReadFloat32() {
+        Span<byte> bytes = stackalloc byte[4] { Data[Index + 3], Data[Index + 2], Data[Index + 1], Data[Index] };
```

Comment:
do you even need stackalloc, you can just `ReadOnlySpan` as a window over the underlying data